### ISSUE TITLE
stdout/stdout buffering on Windows

### ISSUE THREAD

**Issue by CarstenKoenig** [idris-lang/Idris-dev#3483]
the default buffering used in Windows (Win10 64bit / de-DE) will make things like

```
plusIO : IO Int
plusIO = do
  putStr "Inp 1: "
  a <- getLine
  putStr "Inp 2: "
  b <- getLine
  pure $ a ++ b
```

behave strangely (it will only show the prompts after you did input both values)

On top of this it's impossible to use `:x plusIO` from emacs (Atoms REPL won't work at all)

---

This is a problem I have with every console I/O in Haskell and the usual solution is to use `hSetBuffering` to `NoBuffering` or issuing some `hFlush`s in between.

So it would be nice if you could for example add some `hFlush`s (probably around here: https://github.com/idris-lang/Idris-dev/blob/692d61015dc0c7ee8ea1279b77e9a5f339dff3a7/src/Idris/Core/Execute.hs#L452)

(I don't feel competent enough to issue a PR myself  ... but I can try if this is something you would consider)

**Comment by melted** [comment]
I think you should take a shot at it, it's always nice to get new contributors!

I'll do it otherwise, but I'm likely even more clueless about things.

**Comment by CarstenKoenig** [comment]
thanks

as this is probably not very important for most people I would love to give this a try - although it will probably take me a few days

It's probably better to add something similar like `hSetBuffering` for stdout/stdin to `...Interactive` (where `putStr'`, etc. is located) - so I would start with this if this is fine with you.

**Comment by david-christiansen** [comment]
Just so you know, `Execute.hs` is only used by `:x` and the type provider extension. Compiled code doesn't go through that interpreter, and compiled code is probably what you want to update. 

I'd suggest adding the equivalent of `hSetBuffering` using the C FFI, and then calling it in the appropriate places.

**Comment by melted** [comment]
@david-christiansen Note the `:x` in the example.

**Comment by david-christiansen** [comment]
Gah, sorry! Thanks!

**Comment by ahmadsalim** [comment]
Seems that this was resolved by #3495 , so I will close it.
